### PR TITLE
online_status: Add format

### PR DESCRIPTION
**Pull request by lasers**
This adds format: string to print (default '{state}')

Ongoing efforts to `FORMAT ALL THE THINGS!` (meme). https://github.com/ultrabug/py3status/issues/98

EDIT: Srsly. What to do here? .__.
```
online_status.py:64:100: E501 line too long (100 > 99 characters)
* response['full_text'] = self.py3.safe_format(self.format, {'state': self.format_online})

online_status.py:67:100: E501 line too long (101 > 99 characters)
* response['full_text'] = self.py3.safe_format(self.format, {'state': self.format_offline})
```
EDIT: 👍

### CODE REVIEW

**Comment by tobes** [comment]
I understand why you did that last patch but it is bad.

`response` is used commonly across may modules and changing it to `reply` makes the module much harder to work with.

A better option to help with the long lines and I'd say improve readability would just to do something like (this could be done better for this actual module)
```
if ...:
    full_text = ....
    color = ...
else:
    full_text = ....
    color = ...

return {
    'full_text': full_text,
    'color': color,
    'cached_until': ....,
}
```

**Comment by lasers** [comment]
Fixed. Can do `if self._connection_present():` at your request.

File: py3status/modules/online_status.py [hunk 1]
```diff
@@ -50,17 +55,19 @@ def _connection_present(self):
                                    stdout=fnull, stderr=fnull) == 0
 
     def online_status(self):
+
         response = {
-            'cached_until': self.py3.time_in(self.cache_timeout)
+            'cached_until': self.py3.time_in(self.cache_timeout),
+            'full_text': self.py3.safe_format(self.format, {'state': self.format_offline}),
+            'color': self.py3.COLOR_BAD
         }
```
Comment (tobes) [inline py3status/modules/online_status.py]: response is calculated and then just dumped if we are connected.  We should only be calculating the stuff we need to.


`py3.safe_format()` is very expensive so we want to only do it when we need to.  At some point I want to re-write the formatter so it is much more efficient but that is another story.

also 'cached_until' is not part of response when we are connected

**Comment by lasers** [comment]
if final_push:
&nbsp;&nbsp;&nbsp;&nbsp;Thanks @tobes

File: py3status/modules/online_status.py [hunk 2]
```diff
@@ -50,17 +55,18 @@ def _connection_present(self):
                                    stdout=fnull, stderr=fnull) == 0
 
     def online_status(self):
-        response = {
-            'cached_until': self.py3.time_in(self.cache_timeout)
-        }
+        response = {'cached_until': self.py3.time_in(self.cache_timeout)}
```
Comment (tobes) [inline py3status/modules/online_status.py]: cached_until is needed in the actual response

Comment (lasers) [inline py3status/modules/online_status.py]: Sorry. Could you explain more?

Comment (lasers) [inline py3status/modules/online_status.py]: What to change? Fill `cache_until` in two responses? `Line 58` should be fine?

Comment (tobes) [inline py3status/modules/online_status.py]: how does the value in line 58 get returned from the function?

what can happen to response defined in that line?  maybe it gets overwritten?

Comment (lasers) [inline py3status/modules/online_status.py]: Fixed (?). I jump in and try python. Please let me know if that's the wanted behaviour. It looks like `net_rate.py:157` may wrong too if it's going to constantly check things too.

Comment (tobes) [inline py3status/modules/online_status.py]: we are all learning :)

line 58 can just die

**Comment by ultrabug** [comment]
We could argue that the format_offline/online options should be renamed to icon_offline/online just like you did on the DPMS PR but this is cherry on the cake and arguable.

Thanks a lot !

**Comment by tobes** [comment]
> We could argue that the format_offline/online options should be renamed to icon_offline/online just like you did on the DPMS PR

I'd have done that too

**Comment by lasers** [comment]
I'll look at that. Use depreciation stuffs too. No problem.

Today is the day you have more PR than issues. 🔥
Got a question. I don't see an `once` option for cache_timeout? 

I can't output whoami or something else once in `external script` module instead of using `whoami` module.

**Comment by tobes** [comment]
> Got a question. I don't see an once option for cache_timeout?

`cache_until': self.py3.CACHE_FOREVER`

**Comment by lasers** [comment]
What for config file?

**Comment by tobes** [comment]
> What for config file?

That doesn't really make any sense this option is for modules that have constant output, whoami should just cache forever imho

users can still force a refresh by middle clicking unless the module stops them force-ably

**Pull request merged**